medmodels/treatment_effect_estimation/builder.py

```diff
@@ -35,11 +35,12 @@ class TreatmentEffectBuilder:
     filter_controls_operation: Optional[NodeOperation]
 
     matching_method: Optional[MatchingMethod]
-    matching_essential_covariates: Optional[MedRecordAttributeInputList]
-    matching_one_hot_covariates: Optional[MedRecordAttributeInputList]
-    matching_model: Optional[Model]
-    matching_distance_metric: Optional[Metric]
-    matching_number_of_neighbors: Optional[int]
+    matching_essential_covariates: MedRecordAttributeInputList
+    matching_one_hot_covariates: MedRecordAttributeInputList
+    matching_model: Model
+    matching_distance_metric: Metric
+    matching_number_of_neighbors: int
+    matching_nearest_neighbors_algorithm: NNAlgorithm
```

Comment:
if the builder can be configured without these attributes, then they should be optional. This is not the same type as in the treatment effect itself